### ISSUE TITLE
Support generic context type on ContextSetter and setContext

### ISSUE THREAD

**Issue by aaronadamsCA** [apollographql/apollo-client#8902]
It would be nice if your types took context generics, something like:

```diff
diff --git a/node_modules/@apollo/client/link/context/index.d.ts b/node_modules/@apollo/client/link/context/index.d.ts
index f84b24e..8ca0335 100644
--- a/node_modules/@apollo/client/link/context/index.d.ts
+++ b/node_modules/@apollo/client/link/context/index.d.ts
@@ -1,4 +1,4 @@
 import { ApolloLink, GraphQLRequest } from '../core';
-export declare type ContextSetter = (operation: GraphQLRequest, prevContext: any) => Promise<any> | any;
-export declare function setContext(setter: ContextSetter): ApolloLink;
+export declare type ContextSetter<T = any> = (operation: GraphQLRequest, prevContext: T) => Promise<T> | T;
+export declare function setContext<T = any>(setter: ContextSetter<T>): ApolloLink;
 //# sourceMappingURL=index.d.ts.map
\ No newline at end of file
```

You can already manually type input and output contexts, so this is probably more of a nice-to-have, but gradually pushing `any` types out of the `.d.ts` files is probably a good thing.

**Comment by alessbell** [comment]
Thanks to @Hugodby for the PR that addressed this! It's just been merged and will go out in the next release, `3.7.5`, some time next week 🎉

**Comment by alessbell** [comment]
Forgot to close this issue out in January, the fix shipped in `3.7.5` as noted above :^)
